src/ui/common/src/components/resources/dialogs/dialog.tsx

```diff
@@ -108,23 +115,24 @@ const ResourceDialog: React.FC<Props> = ({
     // We do this so we can collect name form inputs inside the same form context.
     const name = data.name;
     // remove the name key from data so pydantic doesn't throw error.
-    delete data.name;
+    const config = { ...data };
```

Comment:
@eunice-chan I've manually included  your change in https://github.com/aqueducthq/aqueduct/pull/1381 here, so feel free to close the PR!